### PR TITLE
switch to offline evaluation paradigm and add save/load

### LINKED ISSUE #71: explore subselection in analysis scripts doesn't make sense anymore
because the approach only 'cares' whether exploration is allowed during replanning. maybe better to explicitly separate evaluation trials...

### PR DESCRIPTION
**Pull request by tomsilver**
tiny

closes #71

### CODE REVIEW

**Comment by tomsilver** [comment]
@RealZiangLiu please check my changes to `run_single_experiment.py` extra carefully because subtle bugs there would be doom

File: experiments/run_single_experiment.py [hunk 1]
```diff
@@ -25,56 +26,138 @@ def _main(cfg: DictConfig) -> None:
...
+            # Save the models from the training approach and load them into the
+            # eval approach.
+            step_model_dir = model_dir / str(t)
+            step_model_dir.mkdir(exist_ok=True)
+            train_approach.save(step_model_dir)
+            eval_approach.load(step_model_dir)
+            # Run evaluation.
+            step_eval_metrics = _evaluate_approach(eval_approach, eval_env, cfg, t)
+            eval_metrics.append(step_eval_metrics)
+        # Eval on the last time step but don't train anymore.
```
Comment (RealZiangLiu) [inline experiments/run_single_experiment.py:96]: Consider adding an `assert cfg.max_environment_steps % cfg.eval_frequency == 0` when loading the config to make sure of this

**Review by RealZiangLiu — APPROVED** [review]

File: experiments/run_single_experiment.py [hunk 2]
```diff
@@ -25,56 +26,138 @@ def _main(cfg: DictConfig) -> None:
...
+def _evaluate_approach(
+    eval_approach: BaseApproach, eval_env: gym.Env, cfg: DictConfig, training_step: int
+) -> dict[str, float]:
+    """Evaluate the given approach and return metrics."""
+    # Evaluate for a given number of trials.
+    cumulative_user_satisfactions: list[float] = []
+    for eval_trial_idx in range(cfg.num_eval_trials):
+        logging.info(f"Starting eval trial {eval_trial_idx}")
+        seed = cfg.seed + cfg.eval_seed_offset + eval_trial_idx
+        obs, info = eval_env.reset(seed=seed)
```
Comment (RealZiangLiu) [inline experiments/run_single_experiment.py:138]: Just making sure, does the code ever rely on sampling directly from `env.action_space`? I didn't see any related usage, but it seems that in the experiment setup, both `train_env.action_space` and `eval_env.action_space` were explicitly seeded, but not here.

**Review by RealZiangLiu — APPROVED** [review]

**Comment by RealZiangLiu** [comment]
> @RealZiangLiu please check my changes to `run_single_experiment.py` extra carefully because subtle bugs there would be doom

Everything looks good except for the two minor checks needing confirmation above : )

**Pull request merged**